plugins/woocommerce/src/Caches/OrderCache.php

```diff
@@ -2,21 +2,29 @@
 
 namespace Automattic\WooCommerce\Caches;
 
-use Automattic\WooCommerce\Caching\CacheException;
 use Automattic\WooCommerce\Caching\ObjectCache;
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
 
 /**
  * A class to cache order objects.
  */
 class OrderCache extends ObjectCache {
 
 	/**
-	 * Get the identifier for the type of the cached objects.
+	 * Get the cache key and prefix to use for Orders.
 	 *
 	 * @return string
 	 */
 	public function get_object_type(): string {
-		return 'orders';
+		if ( 'yes' === get_option( CustomOrdersTableController::HPOS_DATASTORE_CACHING_ENABLED_OPTION ) ) {
+			/**
+			 * The use of datastore caching moves persistent data caching to the datastore. Order object caching then only
+			 * acts as request level caching as the `order_objects` cache group is set as non-persistent.
+			 */
+			return 'order_objects';
```

Comment:
I'm going to update this docblock to remove the reference to hooks/options. The only built in `CacheEngine` implementation is `WPCacheEngine.php` which doesn't have any underlying hooks or option usage.  

I think the compatibility concern is limited.  It would require that an extension both overwrote the cache engine implementation in a way where it depended on the object type returned from the default cache implementation for some behavior but hard coded the previous default in hook usage, or vice versa.